File: fn/src/main/java/org/sqlbuilder/fn/joins/ValenceUnit_AnnoSet.java
package org.sqlbuilder.fn.joins;

import org.sqlbuilder.common.Insertable;
import org.sqlbuilder.fn.objects.AnnotationSet;
import org.sqlbuilder.fn.objects.ValenceUnit;

import java.util.HashSet;
import java.util.Set;

public class ValenceUnit_AnnoSet extends Pair<ValenceUnit, Integer> implements Insertable<ValenceUnit_AnnoSet>
{
	public static final Set<ValenceUnit_AnnoSet> SET = new HashSet<>();

	// C O N S T R U C T O R

	public static ValenceUnit_AnnoSet make(final ValenceUnit vu, final int annosetid)
	{
		var va = new ValenceUnit_AnnoSet(vu, annosetid);
		SET.add(va);
		return va;
	}

	private ValenceUnit_AnnoSet(final ValenceUnit vu, final int annosetid)
	{
		super(vu, annosetid);
	}

	// I N S E R T

	@Override
	public String dataRow()
	{
		return String.format("%s,%s", //
				first.getIntId(), //
				second);
	}

	// T O S T R I N G

	@Override
	public String toString()
	{
		return String.format("[VU-AS vu=%s annoset=%s]", first, second);
	}
}

File: fn/src/main/java/org/sqlbuilder/fn/joins/ValenceUnit_AnnoSet.kt
package org.sqlbuilder.fn.joins

import org.sqlbuilder.common.Insertable
import org.sqlbuilder.fn.objects.ValenceUnit
import java.util.*

data class ValenceUnit_AnnoSet(
    val vu: ValenceUnit,
    val annosetid: Int,
) : Insertable {

    // I N S E R T

    override fun dataRow(): String {
        return "${vu.intId},$annosetid"
    }

    // I D E N T I T Y

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val that = o as ValenceUnit_AnnoSet
        return vu == that.vu && annosetid == that.annosetid
    }

    override fun hashCode(): Int {
        return Objects.hash(vu, annosetid)
    }

    // T O S T R I N G

    override fun toString(): String {
        return "[VU-AS vu=$vu annoset=$annosetid]"
    }

    companion object {

        val SET = HashSet<ValenceUnit_AnnoSet>()

        fun make(vu: ValenceUnit, annosetid: Int): ValenceUnit_AnnoSet {
            val va = ValenceUnit_AnnoSet(vu, annosetid)
            SET.add(va)
            return va
        }
    }
}
